### PR TITLE
👷 🐳 Dockerize Integration Tests and add Jenkinsfile to run Them

### PR DESCRIPTION
**Pull request by heikomat**
## What did you change?

- Dockerized the integration tests.
- Created a Jenkinsfile to run the integration tests.
- Disabled the test timeouts in the "before" and "after" hooks (They were causing errors if the process engine booted to slow).
- Added a database dump so we don't need to run migrations before executing the tests.

## How can others test the changes?

you can see the tests executed by jenkins [here](http://ci.process-engine.io:8080/blue/organizations/jenkins/process-engine_node-lts%2Fconsumer_api_meta/activity). Nothing much you can test yourself here

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.)
- [x] I've merged the `develop` branch into my branch before finishing this PR.
- [x] I've **not added any other changes** than the ones described above.
- [x] I've mentioned all **PRs, which relate to this one**

### CODE REVIEW

**Comment by heikomat** [comment]
process-engine-pro/organization#63

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+  // Some Dockerfiles create volumes using the `VOLUME` command (see https://docs.docker.com/engine/reference/builder/#volume)
+  // running the speedtests creates two dangling volumes. One is from postgres (which contains data), but i don't know about the other one (which is empty)
+  // Dangling volumes are volumes that are not used anywhere. It is safe to remove them.
+  // This removes all dangling volumes
+  sh(script: "docker volume prune --force");
+}
+
+def slack_send_summary() {
+  def cleanedString = testresults.replace('\n', '\\n').replace('"', '\\"');
```
Comment (Paulomart) [inline Jenkinsfile]: `cleaned_string`

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+  if (test_failed == true) {
+    color_string = '"color":"danger"';
+    title_string =  "\"title\":\":boom: Consumer tests for ${env.BRANCH_NAME} failed!\"";
+  }
+
+  slackSend(attachments: "[{$color_string, $title_string, $markdown_string, $result_string, $action_string}]");
+}
+
+def slack_send_testlog() {
```
Comment (Paulomart) [inline Jenkinsfile]: I would suggest using parameters for this function instead of globals

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+  stages {
+    stage('Prepare') {
+      steps {
+        script {
+
+          def first_seven_digits_of_git_hash = env.GIT_COMMIT.substring(0, 8);
+          def safe_branch_name = env.BRANCH_NAME.replace("/", "_");
+          def image_tag = "${safe_branch_name}-${first_seven_digits_of_git_hash}-b${env.BUILD_NUMBER}";
+
+          dbImage       = docker.build("consumertest_db_image:${image_tag}", '--file _integration_tests/Dockerfile.database _integration_tests');
```
Comment (Paulomart) [inline Jenkinsfile]: `db_image`

File: Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+          serverImage   = docker.build("consumertest_server_image:${image_tag}", '--no-cache --file _integration_tests/Dockerfile.tests _integration_tests');
```
Comment (Paulomart) [inline Jenkinsfile]: `server_image`

File: Jenkinsfile [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+          dbImageId     = dbImage.id;
```
Comment (Paulomart) [inline Jenkinsfile]: `db_image_id`

File: Jenkinsfile [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+          serverImageId = serverImage.id;
```
Comment (Paulomart) [inline Jenkinsfile]: `server_image_id`

File: Jenkinsfile [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+
+          dbContainerId = dbImage
```
Comment (Paulomart) [inline Jenkinsfile]: `db_container_id`

File: Jenkinsfile [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+              sh(script: 'while ! pg_isready -U postgres -h db ; do sleep 5; done');
+          }
+        }
+      }
+    }
+    stage('Consumer Tests') {
+      steps {
+        script {
+          // image.inside mounts the current Workspace as the working directory in the container
+          serverImage.inside("--env NODE_ENV=test --env CONFIG_PATH=/usr/src/app/application/config --env datastore__service__data_sources__default__adapter__server__host=db --link ${dbContainerId}:db") {
```
Comment (Paulomart) [inline Jenkinsfile]: Can you split the string into multiple lines for better readability?

File: Jenkinsfile [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+            errorCode = sh(script: "node /usr/src/app/node_modules/.bin/mocha /usr/src/app/test/**/*.js --exit > result.txt", returnStatus: true);
```
Comment (Paulomart) [inline Jenkinsfile]: `error_code`

**Pull request merged**